airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/retriers/default_retrier.py

```diff
@@ -21,8 +68,30 @@ def max_retries(self) -> Union[int, None]:
     def retry_factor(self) -> float:
         return self._retry_factor
 
-    def should_retry(self, response: requests.Response) -> bool:
-        return response.status_code == 429 or 500 <= response.status_code < 600
+    def should_retry(self, response: requests.Response) -> RetryBehavior:
+        """
+        1. If the retry, retry
+        2. Else if ignore, ignore
+        3. else if error, fail
+        4. else: OK!
+        """
+        for retry_filter in self.retry:
+            if retry_filter.matches(response):
+                return self.backoff_time(response)
+        ignore = False
+        for ignore_filters in self.ignore:
+            if ignore_filters.matches(response):
+                ignore = True
+        if ignore:
+            return NonRetriableBehavior.Ignore
+        if response.status_code == 200:
+            return NonRetriableBehavior.Ok
+        return NonRetriableBehavior.Fail
 
     def backoff_time(self, response: requests.Response) -> Optional[float]:
+        for backoff_strategy in self._backoff_strategies:
```

Comment:
iterate over the backoff strategies sequentially. If any of them return a non-None value, return it.
Else, return none